jsonrpc/filter_manager.go

```diff
@@ -146,8 +147,8 @@ type FilterManager struct {
 
 	subscription blockchain.Subscription
 
-	filters map[string]*Filter
-	lock    sync.Mutex
+	filters     map[string]*Filter
+	filtersLock sync.RWMutex
```

Comment:
Honestly, I think it's fine if we embed this instead of having it as a field.

What do you think?

Reply:
I can't agree with that. Embedding mutex to a small struct that has only 1-2 fields would be good.

Reply:
We can probably simplify the FM to have this behavior (be a 1 - 2 field struct).

Resolving as this is not part of this task 🙂